src/connection-manager/agreement.ts

```diff
@@ -195,6 +194,22 @@ class Agreement {
         last_ledger_time: new Date(),
       }
 
+      let server_version
+
+      if (isPreceedingFlagLedger(validation.ledger_index)) {
+        server_version = decodeServerVersion(validation.server_version)
+        const ballot: Ballot = {
+          master_key: validation.master_key,
+          signing_key,
+          ledger_index: Number(validation.ledger_index),
+          amendments: validation.amendments?.join(','),
+          base_fee: validation.base_fee,
```

Comment:
Should this store the current amount for this network when the validation does not contain these properties?

Reply:
Could you explain what you meant by current amount?

Reply:
If they dont vote they are saying they want it to be what the current rate is.  This could be populated here or derived when you ask for the amendment ballots.

Reply:
I see. Yes we can fill in those current rates using data from `ledger` stream